alpenhorn/daemon/update.py

```diff
@@ -1017,12 +1022,33 @@ def update_loop(queue: FairMultiFIFOQueue, pool: WorkerPool | EmptyPool) -> None
             f"{queue.inprogress_size} in-progress on {len(pool)} workers"
         )
 
-        # Avoid looping too fast.
-        remaining = config.config["daemon"]["update_interval"] - loop_time
-        if remaining > 0:
-            global_abort.wait(remaining)  # Stops waiting if a global abort is triggered
+        if once:
+            # If we're in Exit-after-update mode, wait for updates to complete
+            # and then return
+            first_time = True
+            while True:
+                if queue.qsize + queue.inprogress_size + queue.deferred_size == 0:
+                    log.info("Update complete.  Exiting.")
+                    return
+
+                if first_time:
+                    first_time = False
+                    log.info("Waiting for updates to complete.")
```

Comment:
Is the reason for this construction that you don't want the message "Waiting for updates to complete." in the logs if there were none?

Reply:
Yes.  Well, if it didn't wait.